app/javascript/utils/request_helper.js

```diff
@@ -0,0 +1,32 @@
+const getPath = (endpoint, query) => {
+  return `${endpoint}?query=${query}`;
+};
+
+const request = (endpoint) => {
+  let xhr = null; // Hoist this call so that we can abort previous requests.
+
+  return (query, callback) => {
+    if (xhr && xhr.readyState !== XMLHttpRequest.DONE) {
+      xhr.abort();
+    }
+    const path = getPath(endpoint, query);
+
+    xhr = new XMLHttpRequest();
+    xhr.addEventListener("load", (evt) => {
+      let results = [];
+      try {
+        results = JSON.parse(xhr.responseText);
+      } catch (err) {
+        console.error(
+          `Failed to parse results from endpoint ${path}, error is:`,
+          err,
+        );
+      }
+      callback(results);
+    });
+    xhr.open("GET", path);
+    xhr.send();
+  };
+};
```

Comment:
Any reason why we aren't using the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) instead of creating our own wrapper for XMLHttpRequest?

I don't think browser support should be a problem for us. Especially because we're already using ES6 syntax, and I don't think `esbuild` [has been configured](https://github.com/DFE-Digital/itt-mentor-services/blob/f42e69303f7bd251f41ca38048d4af2cb5010cb9/package.json#L11) to transpile that to ES5 for older browsers.

Reply:
This is extracted directly from https://github.com/DFE-Digital/publish-teacher-training/blob/main/app/javascript/utils/request_helper.js
As this was what Publish was doing, I thought it best to copy their implementation.

Reply:
Seems sensible. Let's use that for now then – it's not doing any harm.